include/dts-functions.sh

```diff
@@ -1730,72 +1624,90 @@ send_dts_logs(){
 }
 
 check_if_fused() {
-  file_path="/sys/class/mei/mei0/fw_status"
-
-  if [[ ! -f $file_path ]]; then
-    echo "File not found: $file_path"
+  local _file_path
+  _file_path="/sys/class/mei/mei0/fw_status"
+  local _file_content
+  local _hfsts6_value
+  local _line_number
+  local _hfsts6_binary
+  local _binary_length
+  local _padding
+  local _zeros
+  local _bit_30_value
+
+  if ! $FSREAD_TOOL test -f "$_file_path"; then
+    print_error "File not found: $_file_path"
     return 2
   fi
 
-  hfsts6_value=""
-  line_number=1
+  _file_content="$($FSREAD_TOOL cat $_file_path)"
+
+  _fsts6_value=""
+  _line_number=1
   while IFS= read -r line; do
-    if [[ $line_number -eq 6 ]]; then
-      hfsts6_value=$line
+    if [[ $_line_number -eq 6 ]]; then
+      _hfsts6_value="$line"
       break
     fi
-    ((line_number++))
-  done <"$file_path"
+    ((_line_number++))
+  done <<< "$_file_content"
 
-  if [[ -z $hfsts6_value ]]; then
-    echo "Failed to read HFSTS6 value"
+  if [[ -z "$_hfsts6_value" ]]; then
+    print_error "Failed to read HFSTS6 value"
     exit 1
```

Comment:
Do you really want to exit here instead of returning 1? It would still exit with error in `dasharo-deploy` with
`error_exit "Dasharo cannot be installed on this platform"`